packages/services/src/terminal/restapi.ts

```diff
@@ -120,6 +121,71 @@ export async function shutdownTerminal(
   }
 }
 
+/**
+ * The Terminal API client.
+ *
+ * #### Notes
+ * Use this class to interact with the Jupyter Server Terminal API.
+ * This class adheres to the Jupyter Server API endpoints.
+ */
+export class TerminalAPIClient implements ITerminalAPIClient {
+  /**
+   * Create a new Terminal API client.
+   */
+  constructor(options: { serverSettings?: ServerConnection.ISettings } = {}) {
+    this.serverSettings =
+      options.serverSettings ?? ServerConnection.makeSettings();
+  }
+
+  /**
+   * The server settings for the client.
+   */
+  readonly serverSettings: ServerConnection.ISettings;
+
+  /**
+   * Whether terminals are available.
+   */
+  get isAvailable(): boolean {
+    return isAvailable();
+  }
+
+  /**
+   * Start a new terminal session.
+   *
+   * @param name - The name of the target terminal.
+   *
+   * @param cwd - The path in which the terminal will start.
+   *
+   * @returns A promise that resolves with the session model.
+   */
+  async startNew(name?: string, cwd?: string): Promise<IModel> {
```

Comment:
This is not a strong opinion, but should we use `options?: ITerminalOptions` pattern here? It looks like the caller already does use `ITerminal.IOptions` (we would probably need to make that one inherit from a new interface).

This way all `startNew` methods would follow the `options` API, and adding/removing things is easier.

Reply:
Agree that switching to `options?: ITerminalOptions` brings a couple of advantages 👍 

For kernels, it looks like the manager's `startNew` method currently takes an `IKernelOptions`: 

https://github.com/jupyterlab/jupyterlab/blob/9749b07d54b4ac7622e617f1afdce770849c6f92/packages/services/src/kernel/manager.ts#L179

Defined here:

https://github.com/jupyterlab/jupyterlab/blob/9749b07d54b4ac7622e617f1afdce770849c6f92/packages/services/src/kernel/restapi.ts#L117

So maybe we can follow the same approach have both `startNew` methods take a `ITerminal.IOptions`.